contracts/DCAHubSwapper/ThirdPartyDCAHubSwapper.sol

```diff
@@ -56,6 +57,26 @@ contract ThirdPartyDCAHubSwapper is IDCAHubSwapCallee {
     _handleIntermediateTokens(_callbackData.intermediateTokensToCheck, _callbackData.leftoverRecipient);
   }
 
+  /**
+   * @notice Executed a DCA swap
+   * @dev There are some cases where the oracles differ from what the markets can offer, so a swap can't be executed. But
+   *      it could happen that even if the amounts being swap are really big, the difference between oracle and market is
+   *      only a few dollars. In that case, it would be nice if someone could just pay for the difference.
+   *      The idea here is that instead of calling the hub directly, someone could call the swapper with some native token,
+   *      so that when the swapper gets called, they can use that native token balance as part of the swap, and cover the
+   *      difference
+   */
+  function executeSwap(
+    IDCAHub _hub,
+    address[] calldata _tokens,
+    IDCAHub.PairIndexes[] calldata _pairsToSwap,
+    uint256[] calldata _borrow,
+    bytes calldata _callbackData,
+    bytes calldata _oracleData
+  ) external payable {
+    _hub.swap(_tokens, _pairsToSwap, address(this), address(this), _borrow, _callbackData, _oracleData);
+  }
```

Comment:
This contract should always be approved exact amount by callers, right? If not, then... This function is dangerous since `_hub` can be any contract.

Reply:
There is no approval needed. This function is meant the be used for market swaps + native balance, which doesn't need approval

Reply:
Oh my god, my brain f*cked up. I was thinking of the `CallerOnly`.